action-transfer.Rmd

```diff
@@ -29,50 +32,71 @@ dev.off()
 
 ### Exercise 9.4.2 {-}
 
-Create an app that lets you upload a csv file, select a variable, and then perform a `t.test()` on that variable. After the user has uploaded the csv file, you'll need to use `updateSelectInput()` to fill in the available variables. See Section 10.1 for details.
+Create an app that lets you upload a csv file, select a variable, and then
+perform a `t.test()` on that variable. After the user has uploaded the csv
+file, you'll need to use `updateSelectInput()` to fill in the available
+variables. See Section
+[10.1](https://mastering-shiny.org/action-dynamic.html#updating-inputs)
+for details.
 
 :::solution
 #### Solution {-}
 
-We can use the `fileInput` widget and the `accept` argument to ensure our app only takes on `.csv`s. In the `server` function we'll select the `datapath` from within our file input and use this to update our `selectInput`. We need to put the `updateSelectInput` within a reactive because we need the options to change if the user selects another file. Lastly, we use `input$variable` to create our t-test output.
+We can use the `fileInput` widget with the `accept` argument set to `.csv` to
+allow only the upload of csv files. In the `server` function we save the
+uploaded data to the the `data` reactive and use it to update `input$variable`,
+which displays variable (i.e. data column) choices. Note that we put the
```

Comment:
Totally just nit picking here but do we want to note that we are filtering the columns to only include numeric columns or do you think thats not necessary? Maybe adding the word numeric somewhere like:

> In the server function we save the uploaded data to the the data reactive and use it to update input$variable, which displays variable (i.e. **numeric** data column) choices

Just an idea...